### PR TITLE
Fix coverity leaks 1454341, 1454442, 1456870

### PR DESCRIPTION
**Pull request by amcp**
Use try-with-resources to close streams with a terminal step.

Addresses #790 in part

Signed-off-by: Alexander Patrikalakis <amcp@amazon.co.jp>

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there an issue associated with this PR? Is it referenced in the commit message?
- [x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[skip ci]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchIndex.java [hunk 1]
```diff
@@ -1017,6 +1018,7 @@ public void restore(Map<String,Map<String, List<IndexEntry>>> documents, KeyInfo
             response = client.search(indexStoreName, indexType, compat.createRequestBody(sr, NULL_PARAMETERS), sr.getSize() >= batchSize);
             log.debug("First Executed query [{}] in {} ms", query.getCondition(), response.getTook());
             final ElasticSearchScroll resultIterator = new ElasticSearchScroll(client, response, sr.getSize());
+            // This method returns a Stream. There is no way to return a stream without leaking it. CID 1456881
```
Comment (mbrukman) [inline janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchIndex.java]: `s/CID/Coverity id/` ?

File: janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchIndex.java [hunk 2]
```diff
@@ -1074,6 +1076,7 @@ private ElasticSearchResponse runCommonQuery(RawQuery query, BaseTransaction tx,
         final ElasticSearchResponse response = runCommonQuery(query, tx, size, size >= batchSize );
         log.debug("First Executed query [{}] in {} ms", query.getQuery(), response.getTook());
         final ElasticSearchScroll resultIterator = new ElasticSearchScroll(client, response, size);
+        // This method returns a Stream. There is no way to return a stream without leaking it. CID 1456884
```
Comment (mbrukman) [inline janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchIndex.java]: `s/CID/Coverity id/` ?

File: janusgraph-core/src/main/java/org/janusgraph/diskstorage/keycolumnvalue/scan/StandardScannerExecutor.java [hunk 3]
```diff
@@ -324,10 +324,13 @@ public void run() {
...
                     StaticBuffer key = keyIterator.next();
-                    RecordIterator<Entry> entries = keyIterator.getEntries();
-                    if (!keyFilter.test(key)) continue;
-                    EntryList entryList = StaticArrayEntryList.ofStaticBuffer(entries, StaticArrayEntry.ENTRY_GETTER);
-                    queue.put(new SliceResult(query, key, entryList));
+                    if (!keyFilter.test(key)) {
+                        continue;
+                    }
+                    try (final RecordIterator<Entry> entries = keyIterator.getEntries()) {
+                        queue.put(new SliceResult(query, key,
```
Comment (mbrukman) [inline janusgraph-core/src/main/java/org/janusgraph/diskstorage/keycolumnvalue/scan/StandardScannerExecutor.java]: Maybe keep this as two statements (as it was originally) for readability?

Either way, it's two lines, but the original way breaks it up such that it's easier to read, IMHO.

File: janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchIndex.java (hunk 1 above)
Comment (amcp) [reply janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchIndex.java]: fixed

File: janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchIndex.java (hunk 2 above)
Comment (amcp) [reply janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/ElasticSearchIndex.java]: fixed

File: janusgraph-core/src/main/java/org/janusgraph/diskstorage/keycolumnvalue/scan/StandardScannerExecutor.java (hunk 3 above)
Comment (amcp) [reply janusgraph-core/src/main/java/org/janusgraph/diskstorage/keycolumnvalue/scan/StandardScannerExecutor.java]: set back to two lines and made the entryList final

**Pull request closed without merging**